### DISCUSSION TITLE
There is no permission named `*.create,update,view` for guard `web`.

### DISCUSSION THREAD

**Discussion by sharoonamjid** [spatie/laravel-permission#1927] [Ideas]
After enabling:

`'enable_wildcard_permission' => true,`

I created some permissions:

```
Permission::create(['name' => 'posts.*']);
Permission::create(['name' => 'posts.create']);
Permission::create(['name' => 'posts.update']);
Permission::create(['name' => 'posts.view']);
```

Then created role:

`$subscriber = Role::create(['name' => 'subscriber']);`

Then tried assigning subpath permissions to role:

`$subscriber->givePermissionTo('*.create,update,view');`

but it throws the following Exception:

**There is no permission named `*.create,update,view` for guard `web`.**

It throws the same Exception for user as well:

```
 $user = User::factory()->create();
 $user->givePermissionTo('*.create,update,view');
```

**There is no permission named `*.create,update,view` for guard `web`.**

- spatie/laravel-permission package version: 5.3.1
- laravel/framework version: 8.69.0

**Comment by sharoonamjid** [comment]
Do I need to specifically create permission like below?:

`Permission::create(['name' => '*.create,update,view']);`

It seems to work but I thought It will automatically work by just defining the pattern.

**Comment by erikn69** [comment]
Yes, you have to create the permission as the documentation says
This is not an issue

**Comment by sharoonamjid** [comment]
Thank you. Wouldn't it be nice to define permissions like below:

```
Permission::create(['name' => 'posts.*']);
Permission::create(['name' => 'posts.create']);
Permission::create(['name' => 'posts.update']);
Permission::create(['name' => 'posts.view']);
```

and then this should/could work out of the box:

`$subscriber->givePermissionTo('*.create,update,view');`

OR

`$subscriber->givePermissionTo('posts.create,update,view');`

OR

`$subscriber->givePermissionTo('posts.view,create');`

OR any other combination works out of the box.

**Comment by erikn69** [comment]
If it seems like a good idea, you could do the PR, because no one else will
